scapy/packet.py

```diff
@@ -1555,10 +1620,15 @@ def ls(obj=None, case_sensitive=False, verbose=False):
             pattern = re.compile(obj, 0 if case_sensitive else re.I)
             all_layers = sorted((layer for layer in conf.layers
                                  if (pattern.search(layer.__name__ or '')
-                                     or pattern.search(layer.name or ''))),
+                                     or pattern.search(layer.name if isinstance(layer.name, str) else '' or ''))),
                                 key=lambda x: x.__name__)
         for layer in all_layers:
             print("%-10s : %s" % (layer.__name__, layer._name))
+        if conf.interactive:
+            print()
+            print("TIP: having troubles reading this text ?"
```

Comment:
I think it would be better to add this to the docstring (and users will see that using `help()`). I don't want to be annoyed when I use `ls()` and know what I'm doing.